alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt

```diff
@@ -123,55 +123,30 @@ class TransportExecuteMonitorAction @Inject constructor(
                 }
 
                 if (monitor.monitorType == Monitor.MonitorType.DOC_LEVEL_MONITOR) {
-                    if (!docLevelMonitorQueries.docLevelQueryIndexExists()) {
-                        docLevelMonitorQueries.initDocLevelQueryIndex(object : ActionListener<CreateIndexResponse> {
-                            override fun onResponse(response: CreateIndexResponse) {
+                    scope.launch {
+                        if (!docLevelMonitorQueries.docLevelQueryIndexExists()) {
+                            try {
+                                docLevelMonitorQueries.initDocLevelQueryIndex()
                                 log.info("Central Percolation index ${ScheduledJob.DOC_LEVEL_QUERIES_INDEX} created")
-                                docLevelMonitorQueries.indexDocLevelQueries(
-                                    client,
-                                    monitor,
-                                    monitor.id,
-                                    WriteRequest.RefreshPolicy.IMMEDIATE,
-                                    indexTimeout,
-                                    null,
-                                    actionListener,
-                                    object : ActionListener<BulkResponse> {
-                                        override fun onResponse(response: BulkResponse) {
-                                            log.info("Queries inserted into Percolate index ${ScheduledJob.DOC_LEVEL_QUERIES_INDEX}")
-                                            executeMonitor(monitor)
-                                        }
-
-                                        override fun onFailure(t: Exception) {
-                                            actionListener.onFailure(AlertingException.wrap(t))
-                                        }
-                                    }
-                                )
-                            }
-
-                            override fun onFailure(t: Exception) {
+                            } catch (t: Exception) {
                                 actionListener.onFailure(AlertingException.wrap(t))
+                                return@launch
```

Comment:
you can put this try catch outside the scope of launch and get rid of `return@launch`

Reply:
done